.github/workflows/benchmark-artifact-linux.yml

```diff
@@ -4,63 +4,34 @@ on:
     branches: [ main ]
 jobs:
   build-libraries:
-    runs-on: ubuntu-latest
-    env:
-      CLIENTREPO: ${GITHUB_WORKSPACE}/client
+    runs-on: ubuntu-20.04
     strategy:
-      matrix: 
+      matrix:
         ev-lib: ["", libev, libuv, libevent]
+    env:
+      LIBYAML_VERSION: 0.2.5
     steps:
     - name: Install event libs
-      run: sudo apt-get install -y libevent-dev libev-dev libuv1-dev
-    - name: Download libyaml
-      uses: dsaltares/fetch-gh-release-asset@0.0.5
-      with:
-        repo: "yaml/libyaml"
-        version: "tags/0.2.5"
-        file: yaml-0.2.5.tar.gz
-    - name: Cache libyaml
-      id: cache-libyaml
-      uses: actions/cache@v2
-      env:
-        cache-name: libyaml
-      with:
-        path: yaml-0.2.5
-        key: yaml-0.2.5-${{ runner.os }}
-    - name: Build libyaml
-      id: build-libyaml
-      if: steps.cache-libyaml.outputs.cache-hit != 'true'
       run: |
-        tar xvfz yaml-0.2.5.tar.gz && cd yaml-0.2.5
-        ./configure
-        make
-    - name: install libyaml
-      run: |
-        cd yaml-0.2.5
-        sudo make install
-    - name: Checkout c client
-      uses: actions/checkout@v2
-      with:
-        repository: aerospike/aerospike-client-c
-        submodules: recursive
-        path: client
+        sudo apt-get install -y libevent-dev libev-dev libuv1-dev
     - name: Checkout benchmark
       uses: actions/checkout@v2
       with: 
         path: main
         submodules: recursive
-    - name: Build client lib
-      run: make EVENT_LIB=${{ matrix.ev-lib }}
-      working-directory: client
     - name: Build benchmark
       run: make EVENT_LIB=${{ matrix.ev-lib }}
       working-directory: main
+    - name: Sanity test asbench artifact
+      run: |
+        ./benchmark | grep "ERROR Failed to connect"
+      working-directory: main/target
     - name: Get artifact name
```

Comment:
You could use that conditional action that we use elsewhere to replace the `if then` and `set-output` sytax.  I think it is cleaner